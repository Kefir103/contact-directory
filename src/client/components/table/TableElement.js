import React from 'react';
import { connect } from 'react-redux';

const TableElement = (props) => {
    return (
        <p className={'table-row'}>
            <div className={'table-cell id'}>{props.element.id}</div>
            <div className={'table-cell'}>{props.element.firstName}</div>
            <div className={'table-cell'}>{props.element.lastName}</div>
            <div className={'table-cell'}>{props.element.email}</div>
            <div className={'table-cell'}>{props.element.phone}</div>
        </p>
    );
};

const mapStateToProps = (state, ownProps) => {
    return {};
};

const mapDispatchToProps = (dispatch) => {
    return {};
};

export default connect(mapStateToProps, mapDispatchToProps)(TableElement);
