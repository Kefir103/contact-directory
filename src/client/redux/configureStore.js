import { createStore, applyMiddleware, compose } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension/index';
import { rootReducer } from './reducers/rootReducer';
import thunk from 'redux-thunk';

let store;

export default function configureStore(initialState) {
    if (store) {
        return store;
    }

    store =
        process.env.NODE_ENV === 'production'
            ? createStore(rootReducer, initialState, compose(applyMiddleware(thunk)))
            : createStore(rootReducer, initialState, composeWithDevTools(applyMiddleware(thunk)));

    return store;
}
