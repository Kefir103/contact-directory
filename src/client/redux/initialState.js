export const initialState = {
    filter: {
        sortingMap: new Map([]),
        filterText: '',
        filterFields: [],
        currentPage: 1,
    },
    data: {
        currentElement: {
            elementInfo: undefined,
            elementIndex: -1,
        },
        elements: [],
        currentElements: [],
        filteredElements: undefined,
        inputContainer: {
            isAddButtonDisabled: false,
            isFormOpen: false,
            inputElement: {
                id: 0,
                firstName: '',
                lastName: '',
                email: '',
                phone: '',
            },
            validInputs: {
                id: false,
                firstName: false,
                lastName: false,
                email: false,
                phone: false,
            },
            addingStatus: {
                addingStatusText: '',
                isError: false,
            },
        },
    },
    appStatus: {
        isLoading: false,
        error: undefined,
    },
};
