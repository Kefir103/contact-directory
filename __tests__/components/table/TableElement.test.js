import React from 'react';
import { shallow, render } from 'enzyme';
import configureMockStore from 'redux-mock-store';
import TableElement from '../../../src/client/components/table/TableElement';
import { initialState } from '../../../src/client/redux/initialState';

const mockStore = configureMockStore();

describe('<TableElement />', () => {
    test('should render correctly without currentElement', () => {
        const element = {
            id: 1,
            firstName: 'firstName',
            lastName: 'lastName',
            email: 'email',
            phone: 'phone',
        };
        const elementIndex = 1;
        const store = mockStore({ ...initialState });
        const wrapper = shallow(
            <TableElement store={store} element={element} elementIndex={elementIndex} />
        );
        const component = wrapper.dive().dive();
        expect(component).toMatchSnapshot();
    });
    test('should render correctly with selected element', () => {
        const element = {
            id: 1,
            firstName: 'firstName',
            lastName: 'lastName',
            email: 'email',
            phone: 'phone',
        };
        const elementIndex = 0;
        const store = mockStore({
            ...initialState,
            data: {
                ...initialState.data,
                pageElements: [{ id: 1 }],
                currentElement: {
                    elementIndex: 0,
                    elementInfo: { id: 1 },
                },
            },
        });
        const wrapper = shallow(
            <TableElement store={store} element={element} elementIndex={elementIndex} />
        );
        const component = wrapper.dive().dive();
        expect(component).toMatchSnapshot();
    });
});
