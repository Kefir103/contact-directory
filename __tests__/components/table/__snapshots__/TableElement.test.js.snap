// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TableElement /> should render correctly with selected element 1`] = `
<div
  className="table-row selected-row"
  onClick={[Function]}
>
  <div
    className="table-cell id"
  >
    1
  </div>
  <div
    className="table-cell"
  >
    firstName
  </div>
  <div
    className="table-cell"
  >
    lastName
  </div>
  <div
    className="table-cell"
  >
    email
  </div>
  <div
    className="table-cell"
  >
    phone
  </div>
</div>
`;

exports[`<TableElement /> should render correctly without currentElement 1`] = `
<div
  className="table-row"
  onClick={[Function]}
>
  <div
    className="table-cell id"
  >
    1
  </div>
  <div
    className="table-cell"
  >
    firstName
  </div>
  <div
    className="table-cell"
  >
    lastName
  </div>
  <div
    className="table-cell"
  >
    email
  </div>
  <div
    className="table-cell"
  >
    phone
  </div>
</div>
`;
